edn_format/immutable_list.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import collections
+
+
+class ImmutableList(collections.MutableSequence, collections.Hashable):
+    def __init__(self, wrapped_list, copy=True):
+        self._list = list(wrapped_list) if copy else wrapped_list
```

Comment:
If we use a list, we get the comparison with a regular list on `__eq__` for free. By using a tuple, we'd have to convert every time we compare.